niri-config/src/lib.rs

```diff
@@ -225,6 +225,10 @@ pub struct Touchpad {
     pub middle_emulation: bool,
     #[knuffel(child, unwrap(argument))]
     pub scroll_factor: Option<FloatOrInt<0, 100>>,
+    #[knuffel(child, unwrap(argument))]
+    pub scroll_factor_horizontal: Option<FloatOrInt<-100, 100>>,
+    #[knuffel(child, unwrap(argument))]
+    pub scroll_factor_vertical: Option<FloatOrInt<-100, 100>>,
```

Comment:
It would be more consistent with other options to have these as properties on `scroll-factor`, i.e.:
```kdl
// The combined version
scroll-factor 2
// Split version
scroll-factor horizontal=3 vertical=-1
// Override just one
scroll-factor 1 vertical=-2
```
Should be doable by adding a new `struct ScrollFactor` with one argument and two property fields.

Reply:
Yeah, that does look much clearer.

- Implemented `ScrollFactor` struct with combined and split configuration support
- Added support for your suggested syntax: `scroll-factor 2`, `scroll-factor horizontal=3 vertical=-1`, `scroll-factor 2 vertical=-1` (base + override)